src/colvarcomp_distances.cpp

```diff
@@ -1056,6 +1094,20 @@ void colvar::rmsd::calc_value()
   for (size_t ia = 0; ia < atoms->size(); ia++) {
     x.real_value += ((*atoms)[ia].pos - ref_pos[ia]).norm2();
   }
+  best_perm_index = 0;
+
+  // Compute sum of squares for each symmetry permutation of atoms, keep the smallest
+  size_t ref_pos_index = atoms->size();
+  for (size_t ip = 1; ip < n_permutations; ip++) {
+    cvm::real value = 0.0;
+    for (size_t ia = 0; ia < atoms->size(); ia++) {
+      value += ((*atoms)[ia].pos - ref_pos[ref_pos_index++]).norm2();
+    }
+    if (value < x.real_value) {
+      x.real_value = value;
+      best_perm_index = ip;
+    }
+  }
```

Comment:
Do you see this as potentially useful for larger selections (e.g. macromolecules instead of small molecules)? If so, it may be a good idea to split this up as different CVC instances, each with a different ordered selection and reference coordinates, and select the minimum for computing the colvar.  It probably can't be done with just `customFunction`, but definitely `scriptedFunction`.  The reason I bring it up is that different CVCs can be computed in parallel.

Nothing to gain of course in performance for small groups.

Reply:
That is the main, idea behind the FRMSD code that inspired this (https://www.ncbi.nlm.nih.gov/pubmed/30828834), but ultimately I don't think that is a significant use case for Colvars, because symmetric atoms in macromolecules are anecdotal: most of them are hydrogen, and the others are end groups like carboxylates. So this is intentionally tailored for small molecules with significant symmetry, typically something like phenol.

Actually this can already be achieved with several RMSD CVCs and a `customFunction` that is their min. However that solution:

- is not available in VMD/Dashboard until Lepton gets included
- has several CVCs with the same fittingGroup doing the same alignment
- needs a long configuration string that is mostly duplicated - and DBC config strings are quite long and complex already
- requires the user to manually provide different sets of reference positions

A straightforward extension of the present implementation, however, is to provide alternate ways to define the permutations in the input. I agree that this one is potentially cumbersome. Only mentioning atoms that actually change in the permutation is a good idea.

Reply:
> A straightforward extension

Edit: it's not that straightforward. I will only implement it if a real user requests it.